src/main/java/com/project/serverpool/controller/ServerController.java

```diff
@@ -18,14 +19,19 @@
 @RestController
 @RequestMapping("/server-pool")
 public class ServerController {
-    ServerService serverService = new ServerService();
+    private ServerService serverService ;
 
-    private ModelMapper mapper = new ModelMapper();
+    private ModelMapper mapper ;
 
+    @Autowired
+    public ServerController(ServerService serverService, ModelMapper mapper) {
+        this.serverService = serverService;
+        this.mapper = mapper;
+    }
 
     @GetMapping
```

Comment:
I see that you are not specifying a path value here , it will end up defaulting to an empty path. It is better if you add proper paths.

Reply:
You can refer to this page for example https://www.javadevjournal.com/spring/rest-resource-naming/